### PR TITLE
Add support for Synapse MetaStore

### PR DESCRIPTION
**Pull request by yanjungao718**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by yanjungao718** [comment]
this is based on [pr 16498](https://github.com/Azure/azure-powershell/pull/16498)

File: src/Synapse/Synapse/Commands/DataPlaneCommands/Artifact/Metastore/RegisterAzureSynapseMetastore.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Synapse.Common;
+using Microsoft.Azure.Commands.Synapse.Models;
+using Microsoft.Azure.Commands.Synapse.Properties;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Synapse
```
Comment (wyunchi-ms) [inline src/Synapse/Synapse/Commands/DataPlaneCommands/Artifact/Metastore/RegisterAzureSynapseMetastore.cs:23]: I think `New` is a better verb to keep consistent with the other cmdlets in Az PowerShell.

File: src/Accounts/Accounts/ChangeLog.md [hunk 2]
```diff
@@ -19,6 +19,7 @@
 -->
 
 ## Upcoming Release
+* Upgraded Azure.Core from 1.21.0 to 1.22.0.
```
Comment (wyunchi-ms) [inline src/Accounts/Accounts/ChangeLog.md:22]: Could you please explain the reason of upgrading `Azure.Core`?

**Comment by wyunchi-ms** [comment]
Please update `changelog.md` of `Synapse`

File: src/Accounts/Accounts/ChangeLog.md (hunk 2 above)
Comment (yanjungao718) [reply src/Accounts/Accounts/ChangeLog.md:22]: The Artifact SDK is upgraded to 1.0.0-preview.13, the version of Azure.Core dependency needs to be greater than or equal to 1.22.0, so I upgrade it here.

File: src/Synapse/Synapse/Commands/DataPlaneCommands/Artifact/Metastore/RegisterAzureSynapseMetastore.cs (hunk 1 above)
Comment (yanjungao718) [reply src/Synapse/Synapse/Commands/DataPlaneCommands/Artifact/Metastore/RegisterAzureSynapseMetastore.cs:23]: @wonner please help to determine.

Comment (wonner) [reply src/Synapse/Synapse/Commands/DataPlaneCommands/Artifact/Metastore/RegisterAzureSynapseMetastore.cs:23]: hi @wyunchi-ms , feature team has approved this cmdlet name, so we would like to use `register` here. thanks :)

**Comment by yanjungao718** [comment]
@wyunchi-ms 
The failed checks are due to property value changed in SDK, do you have any idea about how to fix it? Thanks!

**Comment by yanjungao718** [comment]
@VeryEarly 
Hi Yabo, this pr is for syms cmdlets, here is the related [design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1117) .
Now some checks are failed due to property value changed in SDK, do you have any idea about how to fix it? Thanks!

**Comment by yanjungao718** [comment]
close this pull request due to Syms feature is paused according to Trident Priorities.

**Pull request closed without merging**